### PR TITLE
Removes unwanted leading and trailing edges in the Heatmap column labels.

### PR DESCRIPTION
**Pull request by upendrakumbham**
This PR removes unwanted leading and trailing edges in the Heatmap column labels.

### CODE REVIEW

**Review by alfonsomunozpomer — CHANGES_REQUESTED** [review]
Please add tests.

**Comment by upendrakumbham** [comment]
> Please add tests.

Hi @alfonsomunozpomer, Added a test scenario to test the empty factor type values.

File: src/test/java/uk/ac/ebi/atlas/experimentpage/context/BaselineRequestContextTest.java [hunk 1]
```diff
@@ -195,4 +195,43 @@ public void whenSomeTypesAreTheSameAcrossTheSetTheyDoNotGoIntoTheName() {
...
+
+    @Test
+    public void filterOutFactorTypeEmptyValuesTheyDoNotGoIntoTheName() {
+        String defaultQueryFactorType = "defaultQueryFactorType";
+        String otherType = "otherQueryFactorType";
+
+        AssayGroup ag1 = AssayGroupFactory.create("g1", "run11");
+        AssayGroup ag2 = AssayGroupFactory.create("g2", "run21");
+        AssayGroup ag3 = AssayGroupFactory.create("g3", "run22");
+        List<AssayGroup> assayGroups = ImmutableList.of(ag1, ag2, ag3);
```
Comment (alfonsomunozpomer) [inline src/test/java/uk/ac/ebi/atlas/experimentpage/context/BaselineRequestContextTest.java:207]: The class `RandomDataTestUtils` provides methods such as:
- `generateRandomRnaSeqRunId()`
- `generateRandomAssayGroupId()`
- `generateRandomAssayGroup()` 
- `generateRandomAssayGroups(int count)`

Think if you could randomise your test subjects.

**Review by alfonsomunozpomer — CHANGES_REQUESTED** [review]
Please, have a look at the comments I left above for some improvements.

File: src/main/java/uk/ac/ebi/atlas/experimentpage/context/BaselineRequestContext.java [hunk 2]
```diff
@@ -79,6 +80,7 @@ public String displayNameForColumn(AssayGroup assayGroup) {
                     assayGroup,
                     typesWhoseValuesToDisplay().stream()
                             .map(type -> factorGroup.factorOfType(Factor.normalize(type)).getValue())
+                            .filter(StringUtils::isNotBlank)
```
[the remark this answers is not in the document]
Comment (upendrakumbham) [reply src/main/java/uk/ac/ebi/atlas/experimentpage/context/BaselineRequestContext.java:86]: done

File: src/test/java/uk/ac/ebi/atlas/experimentpage/context/BaselineRequestContextTest.java [hunk 3]
```diff
@@ -195,4 +195,43 @@ public void whenSomeTypesAreTheSameAcrossTheSetTheyDoNotGoIntoTheName() {
         assertThat(subject.displayNameForColumn(ag1), (is("name for g1")));
         assertThat(subject.displayNameForColumn(ag2), (is("name for g2")));
     }
+
+    @Test
+    public void filterOutFactorTypeEmptyValuesTheyDoNotGoIntoTheName() {
+        String defaultQueryFactorType = "defaultQueryFactorType";
```
[the remark this answers is not in the document]
Comment (upendrakumbham) [reply src/test/java/uk/ac/ebi/atlas/experimentpage/context/BaselineRequestContextTest.java]: Sorry, fixed it for all the tests.

File: src/test/java/uk/ac/ebi/atlas/experimentpage/context/BaselineRequestContextTest.java (hunk 1 above)
Comment (upendrakumbham) [reply src/test/java/uk/ac/ebi/atlas/experimentpage/context/BaselineRequestContextTest.java]: Randomised all tests how much I can.

**Review by alfonsomunozpomer — APPROVED** [review]
Looks good. Thanks for cleaning the test code.

**Pull request merged**